core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -155,10 +170,16 @@ public void write(Iterator<Product2<K, V>> records) throws IOException {
       writer.commitAndClose();
     }
 
-    partitionLengths =
-      writePartitionedFile(shuffleBlockResolver.getDataFile(shuffleId, mapId));
-    shuffleBlockResolver.writeIndexFile(shuffleId, mapId, partitionLengths);
+    final File tmpDataFile = blockManager.diskBlockManager().createTempShuffleBlock()._2();
+
+    partitionLengths = writePartitionedFile(tmpDataFile);
```

Comment:
the type of tmp block used here doesn't matter, since [`writePartitionedFile`](https://github.com/apache/spark/blob/master/core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java#L182) doesn't go through `blockManager.getDiskWriter`.  However, I'm a bit confused on this worked before.  The [original partition files might be compressed](https://github.com/apache/spark/blob/master/core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java#L141), and those bytes just get copied to the final data file, though data files seem like they are [never compressed](https://github.com/apache/spark/blob/360ed832f5213b805ac28cf1d2828be09480f2d6/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L1165)?